docs/writing-intel-modules.md

```diff
@@ -0,0 +1,248 @@
+        SET
+        n.lastupdated = {UpdateTag},
+        node.field1 = {value1},
+        node.field2 = {value2},
+        ...
+        node.fieldN = {valueN}
+        ```
+
+- To handle relationships in this case,
+
+    - Every `MERGE` query that creates a new relationship should look like this
+
+        ```cypher
+        ON CREATE SET r.firstseen = {UpdateTag}
+        SET
+        r.lastupdated = {UpdateTag}
+        ```
+
+#### Connecting different node types with the _attach pattern
+
+Node connections can be complex.  In many cases we need to connect many different node types together, so we use an
+`_attach` function to manage this.
+
+The best way to explain `_attach` is through an example, like when [we connect GCP instances to their VPCs](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/gcp/compute.py#L439).
+In this case, we create a [helper _attach function](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/gcp/compute.py#L660)
+that accepts the instance's `id` and connects the instance to the VPC using a `MERGE` query.
+
+This pattern can also be seen when [attaching AWS RDS instances to EC2 security groups](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/aws/rds.py#L108).
+
+
+### Cleanup
+
+We have just added new nodes and relationships to the graph, and we have also updated previously-added ones
+by using `MERGE`.  We now need to delete nodes and relationships that no longer exist, and we do this by simply removing
+all nodes and relationships that have `lastupdated` NOT set to the `update_tag` of this current run.
+
+- Delete all old nodes
+
+    You can see this in our [GCP VPCs example](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/data/jobs/cleanup/gcp_compute_vpc_cleanup.json#L4).
+    We run `DETACH DELETE` to delete an old node and disconnect it from all other nodes.
+
+ - Delete all old relationships
+
+    You can see this in the GCP VPC example [here](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/data/jobs/cleanup/gcp_compute_vpc_cleanup.json#L10)
+    and [here](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/data/jobs/cleanup/gcp_compute_vpc_cleanup.json#L16).
+
+    - Q: We just `DETACH DELETE`'d the node.  Why do we need to delete the relationships too?
+
+    - A: There are cases where the node may continue to exist but the relationships between it and other nodes have changed.
+        Explicitly deleting stale relationships accounts for this case.
+        See this [short discussion](https://github.com/lyft/cartography/pull/124/files#r312277725).
+
+
+
+
+## Error handling principles
+
+- Don't catch the base Exception class when error handling because it makes it difficult to trace.
```

Comment:
Also: try to catch the narrowest possible class of exception. Only catch exceptions when your code can resolve  the issue. Otherwise, allow exceptions to bubble up.